types/types.go

```diff
@@ -149,6 +150,8 @@ type ImageDestination interface {
 	SupportsSignatures() error
 	// ShouldCompressLayers returns true iff it is desirable to compress layer blobs written to this destination.
 	ShouldCompressLayers() bool
+	// ShouldDecompressLayers returns true iff it is desirable to decompress layer blobs written to this destination.
+	ShouldDecompressLayers() bool
```

Comment:
Please use a single `enum` (with cases compress/decompress/preserveoriginal) instead of two booleans, to make structurally sure `ShouldCompressLayers`&&`ShouldDecompressLayers` can’t be true at the same time.

Reply:
Ok.